File: fix_reset.js
const fs = require('fs');
const path = 'd:\\HỌC TỪ VỰNG\\index.html';
let content = fs.readFileSync(path, 'utf8');

const old = "document.getElementById('completion-overlay').remove(); selectDay(curLevel, curType, curDay);\"";
const rep = "document.getElementById('completion-overlay').remove(); renderQuiz(); if(typeof resetTimer==='function') resetTimer(); window.scrollTo({top:0,behavior:'smooth'});\"";

if (content.includes(old)) {
    content = content.replace(old, rep);
    fs.writeFileSync(path, content, 'utf8');
    console.log('Done! Replaced successfully.');
} else {
    // Search for partial match
    const idx = content.indexOf('selectDay(curLevel, curType, curDay)');
    if (idx >= 0) {
        console.log('Found partial at char:', idx);
        console.log('Context:', JSON.stringify(content.slice(idx - 80, idx + 80)));
    } else {
        console.log('Pattern not found. Searching for completion-overlay...');
        const matches = [...content.matchAll(/completion-overlay.*?remove\(\)[^`]{0,150}/g)];
        matches.forEach((m, i) => console.log(i, JSON.stringify(m[0].slice(0, 200))));
    }
}

File: fix_reset.py
import re

path = r'd:\HỌC TỪ VỰNG\index.html'
with open(path, 'r', encoding='utf-8') as f:
    content = f.read()

# Old: selectDay(curLevel, curType, curDay); inside Làm lại từ đầu button
old = 'document.getElementById(\'completion-overlay\').remove(); selectDay(curLevel, curType, curDay);"'
new = 'document.getElementById(\'completion-overlay\').remove(); renderQuiz(); if(typeof resetTimer===\'function\') resetTimer(); window.scrollTo({top:0,behavior:\'smooth\'});"'

if old in content:
    content = content.replace(old, new, 1)
    with open(path, 'w', encoding='utf-8') as f:
        f.write(content)
    print('Done! Replaced successfully.')
else:
    # Try to find what's there
    idx = content.find('selectDay(curLevel, curType, curDay)')
    if idx >= 0:
        print('Found at char:', idx)
        print('Context:', repr(content[idx-50:idx+100]))
    else:
        print('selectDay(curLevel, curType, curDay) not found')
        idx2 = content.find('completion-overlay')
        print('completion-overlay at:', idx2)
